### ISSUE TITLE
DateTimeOffset is bind from Query when using [Header] Attribute. 

### ISSUE THREAD

**Issue by Bartmax** [aspnet/AspNetCore#11927]
### Describe the bug
`DateTimeOffset` (nullabe or not) does not bind from header but from query when using `FromHeader` attribute. 

_**Note: `DateTime` works fine**_

### To Reproduce
1. Create an action with:
```C#
public ActionResult Get([FromHeader]DateTimeOffset dt)
{
    return Ok(dt);
}
```
2. Create a request with header and/or query key dt with value of `2019-01-01`
4. The value from header is never used.

### Expected behavior
dt should have the value from the header.

### Screenshots
![image](https://user-images.githubusercontent.com/3386797/60738770-e07ecc00-9f35-11e9-9c45-3d75def2d3d8.png)


### Additional context
Include the output of `dotnet --info`


 dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100-preview6-012264
 Commit:    be3f0c1a03

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18362
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.0.100-preview6-012264\

Host (useful for support):
  Version: 3.0.0-preview6-27804-01                                                                                        Commit:  fdf81c6faf                                                                                                                                                                                                                           .NET Core SDKs installed:                                                                                                 2.1.604 [C:\Program Files\dotnet\sdk]                                                                                   2.1.700 [C:\Program Files\dotnet\sdk]                                                                                   2.1.800-preview-009677 [C:\Program Files\dotnet\sdk]                                                                    2.1.800-preview-009696 [C:\Program Files\dotnet\sdk]                                                                    2.2.204 [C:\Program Files\dotnet\sdk]                                                                                   2.2.300 [C:\Program Files\dotnet\sdk]                                                                                   2.2.400-preview-010195 [C:\Program Files\dotnet\sdk]                                                                    2.2.400-preview-010219 [C:\Program Files\dotnet\sdk]                                                                    3.0.100-preview5-011568 [C:\Program Files\dotnet\sdk]                                                                   3.0.100-preview6-012264 [C:\Program Files\dotnet\sdk]                                                                                                                                                                                         .NET Core runtimes installed:                                                                                             Microsoft.AspNetCore.All 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]                               Microsoft.AspNetCore.All 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]                                Microsoft.AspNetCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]                               Microsoft.AspNetCore.App 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]                                Microsoft.AspNetCore.App 3.0.0-preview6.19307.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]               Microsoft.NETCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                                     Microsoft.NETCore.App 2.2.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                                      Microsoft.NETCore.App 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                                      Microsoft.NETCore.App 3.0.0-preview6-27804-01 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]                    Microsoft.WindowsDesktop.App 3.0.0-preview6-27804-01 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @Bartmax.
@pranavkm can you please look into this? Thanks!

**Comment by ajaybhargavb** [comment]
@Bartmax, I tried this out with the latest sdk (3.0.100-preview8-013656) and I couldn't reproduce your issue. The values were correctly bound from the sources specified by the attribute (FromQuery or FromHeader) and I saw no inconsistencies between DateTime and DateTimeOffset. Please re-open if you are still seeing the issue.

**Comment by Bartmax** [comment]
Here's a repro:

https://github.com/Bartmax/DateTimeOffsetBindBug


![image](https://user-images.githubusercontent.com/3386797/63895853-bb06bf80-c9c6-11e9-8e69-1b73c210134f.png)

**Comment by Bartmax** [comment]
can't reopen @ajaybhargavb

**Comment by pranavkm** [comment]
@Bartmax setting a compatibility version of 2.1 or later fixes the issue. As @ajaybhargavb noted, the issue does not reproduce with 3.0 either.

**Comment by ajaybhargavb** [comment]
@Bartmax thanks for the repro. As @pranavkm mentioned this is fixed in 3.0 and you can set the compatibility version to get around the issue.

**Comment by Bartmax** [comment]
Excellent thanks!
